spectral_connectivity/wrapper.py

```diff
@@ -25,26 +25,39 @@ def connectivity_to_xarray(m, method='coherence_magnitude', signal_names=None,
         Only makes sense for one pair of signals and symmetrical measures
 
     """
-    if ((method in ['power', 'group_delay', 'canonical_coherence']) or ('directed' in method)):
+    if (method in ['group_delay', ]) or ('directed' in method):
         raise NotImplementedError(f'{method} is not supported by xarray interface')
+    # Name the source and target axes
+    if signal_names is None:
+        signal_names = np.arange(m.time_series.shape[-1])
     connectivity = Connectivity.from_multitaper(m)
     if method == 'canonical_coherence':
-        connectivity_mat, labels = getattr(connectivity, method)(**kwargs)
+        if 'group_labels' not in kwargs:
+            connectivity_mat, labels = getattr(connectivity, method)(group_labels=signal_names,**kwargs)
```

Comment:
I think this should just throw an error because there's no benefit to having the group labels be the signal names (see below for an explanation).